server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationStrategy.scala

```diff
@@ -40,8 +40,30 @@ import scala.jdk.CollectionConverters._
 import scala.util.{Failure, Success, Try}
 
 object UserCredential {
-  type BasicAuthenticationHeaderValue = String Refined MatchesRegex["Basic [\\d\\w=]++"]
-  type CredentialsAsString = String Refined MatchesRegex[".*:.*"]
+  type BasicAuthenticationHeaderValue = String Refined BasicAuthenticationHeaderValueConstraint
+
+  type CredentialsAsString = String Refined CredentialsAsStringConstraint
+
+  case class CredentialsAsStringConstraint()
+  case class BasicAuthenticationHeaderValueConstraint()
+
+  private val charMatcher: CharMatcher = CharMatcher.inRange('a', 'z')
+    .or(CharMatcher.inRange('0', '9'))
+    .or(CharMatcher.inRange('A', 'Z'))
+    .or(CharMatcher.is('_'))
+    .or(CharMatcher.is('='))
+    .or(CharMatcher.is('-'))
+    .or(CharMatcher.is('#'))
+
+  implicit val validateCredentialsAsString: Validate.Plain[String, CredentialsAsStringConstraint] =
+    Validate.fromPredicate(s => s.contains(':'),
+      _ => "Credential string must contains ':'",
+      CredentialsAsStringConstraint())
+
+  implicit val validateBasicAuthenticationHeaderValueConstraint: Validate.Plain[String, BasicAuthenticationHeaderValueConstraint] =
+    Validate.fromPredicate(s => s.startsWith("Basic ") && charMatcher.matchesAllOf(s.substring(6)),
+      _ => "Must start with 'Basic' prefix then be only letter or digits",
```

Comment:
'Basic' or 'Basic ' 
Ignore if it is unnecessary :D

Reply:
'Basic ' the space is important ;-)